### PR TITLE
Make Kotlin do language injections into Java annotations

### PR DESCRIPTION
**Pull request by xiexed**

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjector.kt [hunk 1]
```diff
@@ -247,6 +252,19 @@ class KotlinLanguageInjector(
         return null
     }
 
+    private fun injectInAnnotationCall(host: KtElement): InjectionInfo? {
+        val argument = host.parent as? KtValueArgument ?: return null
+        val annotationEntry = argument.parent.parent as? KtAnnotationEntry ?: return null
+        val callDescriptor = annotationEntry.getResolvedCall(annotationEntry.analyze())?.candidateDescriptor
```
[the remark this answers is not in the document]
Comment (xiexed) [reply idea/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjector.kt]: Is there a simple way to get FQN of annotation without `getResolvedCall()` then ?

[the remark this answers is not in the document]
Comment (yole) [reply idea/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjector.kt]: FQN no, but you can check the short name.

[the remark this answers is not in the document]
Comment (xiexed) [reply idea/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjector.kt]: And what about import aliases then?

[the remark this answers is not in the document]
Comment (yole) [reply idea/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjector.kt]: You can use the PSI to check if they're present in the containing file.

[the remark this answers is not in the document]
Comment (xiexed) [reply idea/src/org/jetbrains/kotlin/idea/injection/KotlinLanguageInjector.kt]: Ok. so there is no simple way ))

**Comment by xiexed** [comment]
Optimization now looks better but will not work without appropriate changes in the Platform which are not yet in master (

**Pull request merged**